File: LeetCode/1-100/80. 删除有序数组中的重复项 II.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  const n = nums.length
  if (n < 3) return n
  let fast = 2,
    slow = 2
  while (fast < n) {
    console.log(nums[slow - 2], nums[fast], slow, fast)
    if (nums[slow - 2] !== nums[fast]) {
      nums[slow] = nums[fast]
      slow++
    }
    fast++
  }
  return slow
}

console.log(removeDuplicates([1, 2, 2, 3]))

File: LeetCode/1-100/80. 删除有序数组中的重复项 II.cpp
class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        if (nums.size() < 2)
            return 1;
        int l = 2;
        for (int i = 2; i < nums.size(); i++)
        {
            if (nums[l - 2] != nums[i])
            {
                nums[l++] = nums[i];
            }
        }
        return l;
    }
};